kyo-combinators/shared/src/main/scala/kyo/Constructors.scala

```diff
@@ -29,7 +29,7 @@ extension (kyoObject: Kyo.type)
       * @return
       *   An effect that ensures the finalizer is executed when the effect is completed
       */
-    def addFinalizer(finalizer: => Unit < IO)(using Frame): Unit < (Resource & IO) =
+    def addFinalizer(finalizer: => Unit < Async)(using Frame): Unit < (Resource & IO) =
```

Comment:
Wouldn't this make more sense as an extension method?

Reply:
I've added `ensuring` like ZIO